c/tree.c

```diff
@@ -633,34 +645,147 @@ LOCAL void alias_ipv4_networks(MMDBW_tree_s *tree)
+// information about such networks.
+LOCAL void insert_reserved_networks_as_fixed_empty(MMDBW_tree_s *tree)
+{
+    insert_networks_as_fixed_empty(tree, reserved_networks_ipv4,
+                                   sizeof(reserved_networks_ipv4) /
+                                   sizeof(struct network));
+
+    if (tree->ip_version == 6) {
+        insert_networks_as_fixed_empty(tree, reserved_networks_ipv6,
+                                       sizeof(reserved_networks_ipv6) /
+                                       sizeof(struct network));
+    }
+}
+
+// Insert a FIXED_EMPTY record for each network.
+LOCAL void insert_networks_as_fixed_empty(MMDBW_tree_s *tree,
+                                          struct network const *const networks,
+                                          const size_t num_networks)
+{
+    for (size_t i = 0; i < num_networks; i++) {
+        MMDBW_network_s resolved_network = resolve_network(tree,
+                                                           networks[i].ipstr,
+                                                           networks[i].prefix_length);
+
+        MMDBW_record_s record = {
+            .type = MMDBW_RECORD_TYPE_FIXED_EMPTY,
+        };
+
+        MMDBW_status const status = insert_record_for_network(tree,
+                                                              &resolved_network,
+                                                              &record,
+                                                              MMDBW_MERGE_STRATEGY_NONE,
+                                                              true);
+
+        free_network(&resolved_network);
+
+        if (MMDBW_SUCCESS != status) {
+            croak(
+                "Unexpected error inserting fixed empty record for network: %s/%u: %s",
+                networks[i].ipstr, networks[i].prefix_length,
+                status_error_message(status));
+        }
+    }
+}
+
 LOCAL MMDBW_status insert_record_for_network(
     MMDBW_tree_s *tree,
     MMDBW_network_s *network,
     MMDBW_record_s *new_record,
     MMDBW_merge_strategy merge_strategy,
     bool is_internal_insert)
 {
-
-    bool is_ipv6 = tree->ip_version == 6;
-    char address_string[ is_ipv6 ? INET6_ADDRSTRLEN : INET_ADDRSTRLEN];
-    inet_ntop(is_ipv6 ? AF_INET6 : AF_INET,
-              network->bytes,
-              address_string,
-              sizeof(address_string));
-
     if (merge_strategy == MMDBW_MERGE_STRATEGY_UNKNOWN) {
         merge_strategy = tree->merge_strategy;
     }
 
     return insert_record_into_next_node(
-               tree,
-               &(tree->root_record),
-               network,
-               0,
-               new_record,
-               merge_strategy,
-               is_internal_insert
-               );
+        tree,
+        &(tree->root_record),
+        network,
+        0,
+        new_record,
+        merge_strategy,
+        is_internal_insert
+        );
```

Comment:
Is the indentation change from running uncrustify?

Reply:
Yeah. Can't say I'm a fan